src/posit/connect/external/databricks.py

```diff
@@ -66,23 +306,107 @@ def sql_credentials_provider(self, *args, **kwargs):
         return lambda: self.__call__(*args, **kwargs)
 
     def auth_type(self) -> str:
-        """Returns the auth type currently in use.
+        return _get_auth_type(self._local_strategy.auth_type())
+
+    def __call__(self, *args, **kwargs) -> CredentialsProvider:
+        # If the content is not running on Connect then fall back to local_strategy
+        if is_local():
+            return self._local_strategy(*args, **kwargs)
+
+        if self._client is None:
+            self._client = Client()
+
+        return PositContentCredentialsProvider(self._client)
+
+
+class PositCredentialsStrategy(CredentialsStrategy):
+    """`CredentialsStrategy` implementation which supports interacting with Viewer OAuth integrations on Connect.
+
+    This strategy callable class returns a `PositCredentialsProvider` when hosted on Connect, and
+    its `local_strategy` strategy otherwise.
+
+    Examples
+    --------
+    NOTE: In the example below, the PositCredentialsProvider *must* be initialized within the context of the 
+    shiny `server` function, which provides access to the HTTP session headers.
 
-        The databricks-sdk client uses the configurated auth_type to create
-        a user-agent string which is used for attribution. We should only
-        overwrite the auth_type if we are using the PositCredentialsStrategy (non-local),
-        otherwise, we should return the auth_type of the configured local_strategy instead
-        to avoid breaking someone elses attribution.
+    ```python
+    import os
 
-        https://github.com/databricks/databricks-sdk-py/blob/v0.29.0/databricks/sdk/config.py#L261-L269
+    import pandas as pd
+    from databricks import sql
+    from databricks.sdk.core import ApiClient, Config, databricks_cli
+    from databricks.sdk.service.iam import CurrentUserAPI
+    from posit.connect.external.databricks import PositCredentialsStrategy
+    from shiny import App, Inputs, Outputs, Session, render, ui
+    
+    DATABRICKS_HOST = "<REDACTED>"
+    DATABRICKS_HOST_URL = f"https://{DATABRICKS_HOST}"
+    SQL_HTTP_PATH = "<REDACTED>"
 
-        NOTE: The databricks-sql client does not use auth_type to set the user-agent.
-        https://github.com/databricks/databricks-sql-python/blob/v3.3.0/src/databricks/sql/client.py#L214-L219
+    app_ui = ui.page_fluid(ui.output_text("text"), ui.output_data_frame("result"))
+
+
+    def server(i: Inputs, o: Outputs, session: Session):
+
+        # HTTP session headers are available in this context.
+        session_token = session.http_conn.headers.get("Posit-Connect-User-Session-Token")
+        posit_strategy = PositCredentialsStrategy(
+            local_strategy=databricks_cli, user_session_token=session_token
```

Comment:
should this now be the patched local strategy?

Reply:
its sort of ambiguous imo. the user could still use the databricks_cli as the local strategy, if say for example their databricks admin doesn't want to hand out service principal credentials. They can still use their own identity for local development with the databricks cli - that was my thinking for keeping this as-is.

Reply:
Added some comments to the docstring to clarify the various use cases.

Reply:
ah good point!